x-pack/plugins/security/server/deprecations/kibana_dashboard_only_role.ts

```diff
@@ -0,0 +1,244 @@
+            }
+          ),
+        ],
+      },
+    },
+  ];
+}
+
+async function getRoleMappingsDeprecations(
+  client: ElasticsearchClient,
+  logger: Logger,
+  packageInfo: PackageInfo
+): Promise<DeprecationsDetails[]> {
+  let roleMappings: SecurityGetRoleMappingResponse;
+  try {
+    roleMappings = (await client.security.getRoleMapping()).body;
+  } catch (err) {
+    logger.error(`Failed to retrieve role mappings: ${getDetailedErrorMessage(err)}`);
+    return deprecationError(packageInfo, err);
+  }
+
+  const roleMappingsWithKibanaDashboardOnlyRole = Object.entries(roleMappings)
+    .filter(([, roleMapping]) => roleMapping.roles.includes(KIBANA_DASHBOARD_ONLY_USER_ROLE_NAME))
+    .map(([mappingName]) => mappingName);
+  if (roleMappingsWithKibanaDashboardOnlyRole.length === 0) {
+    return [];
+  }
+
+  return [
+    {
+      title: i18n.translate(
+        'xpack.security.deprecations.kibanaDashboardOnlyUser.deprecationTitle',
+        {
+          defaultMessage: 'The "{roleName}" role is removed',
+          values: { roleName: KIBANA_DASHBOARD_ONLY_USER_ROLE_NAME },
+        }
+      ),
+      message: i18n.translate(
+        'xpack.security.deprecations.kibanaDashboardOnlyUser.roleMappingsDeprecationMessage',
+        {
+          defaultMessage:
+            'The following role mappings map to a deprecated and removed "{roleName}" role: {roleMappings}. Create a custom role with Kibana privileges to restrict access to just the Dashboard feature instead.',
+          values: {
+            roleName: KIBANA_DASHBOARD_ONLY_USER_ROLE_NAME,
+            roleMappings: roleMappingsWithKibanaDashboardOnlyRole.join(', '),
+          },
+        }
+      ),
+      level: 'warning',
+      deprecationType: 'feature',
+      documentationUrl: `https://www.elastic.co/guide/en/kibana/${packageInfo.branch}/kibana-privileges.html`,
+      correctiveActions: {
+        manualSteps: [
+          i18n.translate(
+            'xpack.security.deprecations.kibanaDashboardOnlyUser.roleMappingsDeprecationCorrectiveActionOne',
+            {
+              defaultMessage:
+                'Create a custom role with Kibana privileges to restrict access to just the Dashboard feature.',
+            }
+          ),
+          i18n.translate(
+            'xpack.security.deprecations.kibanaDashboardOnlyUser.roleMappingsDeprecationCorrectiveActionTwo',
+            {
+              defaultMessage:
+                'Update all role mappings that map to "{roleName}" role to use the custom role instead.',
+              values: { roleName: KIBANA_DASHBOARD_ONLY_USER_ROLE_NAME },
+            }
+          ),
+        ],
+      },
+    },
+  ];
+}
+
+function deprecationError(packageInfo: PackageInfo, error: Error): DeprecationsDetails[] {
+  const title = i18n.translate(
+    'xpack.security.deprecations.kibanaDashboardOnlyUser.deprecationTitle',
+    {
+      defaultMessage: 'The "{roleName}" role is removed',
```

Comment:
__note:__ I use the same title for all cases, including `fetch_error`. It feels reasonable to me since we know what deprecation exactly we want to handled, but I'm not sure if we have any best practices here.